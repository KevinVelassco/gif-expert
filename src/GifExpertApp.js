import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories }) => {

    const [categories, setCategoties] = useState(defaultCategories);

    return (
        <main>
            <header>
                <h1 className="title animate__animated animate__flash">GifExpert</h1>
                <AddCategory setCategories={setCategoties} />
            </header>
            <section>
                {
                    categories.map(category => (
                        <GifGrid key={category} category={category} />
                    ))
                }
            </section>
        </main>
    );
}

GifExpertApp.propTypes = {
    defaultCategories: PropTypes.array
}

GifExpertApp.defaultProps = {
    defaultCategories: ['anime']
}
