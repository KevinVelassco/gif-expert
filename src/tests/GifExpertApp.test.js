import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Test component <GifExpertApp />', () => {
    test('Debe mostrar correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar una lista de categorias', () => {
        const categories = ['zoro', 'goku'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});