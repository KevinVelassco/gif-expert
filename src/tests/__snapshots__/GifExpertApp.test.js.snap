// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifExpertApp /> Debe mostrar correctamente 1`] = `
<main>
  <header>
    <h1
      className="title animate__animated animate__flash"
    >
      GifExpert
    </h1>
    <AddCategory
      setCategories={[Function]}
    />
  </header>
  <section>
    <GifGrid
      category="anime"
      key="anime"
    />
  </section>
</main>
`;

exports[`Test component <GifExpertApp /> debe mostrar una lista de categorias 1`] = `
<main>
  <header>
    <h1
      className="title animate__animated animate__flash"
    >
      GifExpert
    </h1>
    <AddCategory
      setCategories={[Function]}
    />
  </header>
  <section>
    <GifGrid
      category="zoro"
      key="zoro"
    />
    <GifGrid
      category="goku"
      key="goku"
    />
  </section>
</main>
`;
